src/registry/routes/component-info.js

```diff
@@ -4,61 +4,77 @@ var parseAuthor = require('parse-author');
 var _ = require('underscore');
 
 var urlBuilder = require('../domain/url-builder');
+var getComponentInfoOrPreviewFallback = require('./helpers/get-component-fallback').getComponentInfoOrPreviewFallback;
 
-module.exports = function(repository){
-  return function(req, res){
+function getParams(component) {
+  var params = {};
+  if(!!component.oc.parameters){
+    var mandatoryParams = _.filter(_.keys(component.oc.parameters), function(paramName){
+      var param = component.oc.parameters[paramName];
+      return !!param.mandatory && !!param.example;
+    });
+
+    params = _.mapObject(_.pick(component.oc.parameters, mandatoryParams), function(param){
+      return param.example;
+    });
+  }
 
-    repository.getComponent(req.params.componentName, req.params.componentVersion, function(err, component){
+  return params;
+}
 
-      if(err){
-        res.errorDetails = err;
-        return res.status(404).json({ err: err });
+function getParsedAuthor(component) {
+  var author = component.author || {};
+  return _.isString(author) ? parseAuthor(author) : author;
+}
+
+function addGetRepositoryUrlFunction(component) {
```

Comment:
I've split this function from the main callback, because it seemed little bit hard to read.

This kind of refactor probably shouldn't be in this PR, so if it is prepared to revert this change I can make a commit for that.